### DISCUSSION TITLE
EELS fine structure fitting using two gaussians

### DISCUSSION THREAD

**Discussion by CausticChaos** [hyperspy/exspy#63] [Q&A] [answered]
#### Describe the bug

I am trying to analyse a core-loss EELS line-spectrum. I have attempted to fit the data with a series of Gaussian and 1st-order polynomial functions. I have attempted to fit each peak by constraining the fitting of each Gaussian by defining bmin, bmax bounds as well as providing an initial estimate of the Gaussian centre value (corresponding approximately to the specific peak of interest).

I have included code below that attempts to fit a single Gaussian (both Gaussian and GaussianHF have been attempted) to the data but it results in a double-peaked Gaussian in the resulting fitted curve, as illustrated at the end of this bug report. Moreover, the model component parameters do not adjust once the multi-fit is run and the Gaussian curves extend beyond the defined bmin and bmax bounds. No errors have been produced.

I am unsure whether these errors have arisen due to human error or a bug with Hyperspy. However, I have been prompted to submit a bug report by my supervisor.

#### To Reproduce

```
#Define regions for Gaussian fit
region_G1 = [0, 32] 

#Create the model
m = hl_smoothed.create_model(low_loss=ll)

#Remove power law component from the model
m.remove(0)

#Add desired component(s) to model
G1 = hs.model.components1D.Gaussian()

#Extend the model with the components
m.extend([G1])

#Set the regions for Gaussian components
G1.centre.value = (region_G1[0] + region_G1[1]) / 2
G1.centre.bmin = region_G1[0]
G1.centre.bmax = region_G1[1]

#Perform the fit
m.multifit(kind='smart')

#Plot the results
m.plot(plot_components=True)
```

#### Python environment:
 - HyperSpy version: 2.1.0
 - Python version: 3.11.9

#### Additional context

Attached below are illustrations of the code and a figure showing the double-peak Gaussian fit. 

![Code](https://github.com/hyperspy/hyperspy/assets/173935063/7595305a-8944-481b-bbda-5bf1cc8cdc42)

![Figure_EELS_HL_SI_Signal](https://github.com/hyperspy/hyperspy/assets/173935063/79940c96-5517-445a-b65b-efa81a1850b4)

**Comment by ericpre** [comment, marked as answer]
I moved this to a discussion because it seems to be a matter of how it should be use rather than a bug itself.

The approach on estimating sensible starting parameters, setting bounds generally works in these case, even if this is not the most robust and sometimes it is a matter of finding the right way to get the fit to converge to right minima: in this case, it seems to converge in a local minima which is not correct. To model the continuum, usually arctan or core edge components are used instead of polynomial. See for example discussion in https://github.com/hyperspy/exspy/discussions/58

Since there is no error and the data are missing to reproduce what you are seeing, it is difficult to be sure, but I would do the following:
- inspect what the values of the parameters are and see the initial estimate can be improve
- use [`plot(plot_components=True)`](https://hyperspy.org/hyperspy-doc/current/reference/base_classes/model/model1d.html#hyperspy.models.model1d.Model1D.plot) to visualise the contribution of the various components.
- it is possible that you haven't specified to use bounds [`m.fit(bounded=True)`](https://hyperspy.org/hyperspy-doc/current/reference/base_classes/model/basemodel.html#hyperspy.model.BaseModel.fit) or [`m.multifit(bounded=True)`](https://hyperspy.org/hyperspy-doc/current/reference/base_classes/model/basemodel.html#hyperspy.model.BaseModel.multifit). Note that not all optimization algorithms support bounds: https://hyperspy.org/hyperspy-doc/current/user_guide/model/fitting_model.html#fitting-the-model-to-the-data

It would be useful if your supervisor could give this kind of generic advice! 😉
